ReviewBot.py

```diff
@@ -375,6 +379,63 @@ def set_request_ids_project(self, project, typename):
             req.read(request)
             self.requests.append(req)
 
+    def comment_handler_add(self, level=logging.INFO):
+        """Add handler to start recording log messages for comment."""
+        self.comment_handler = CommentFromLogHandler(level)
+        self.logger.addHandler(self.comment_handler)
+
+    def comment_handler_remove(self):
+        self.logger.removeHandler(self.comment_handler)
+
+    def comment_find(self, request=None, state=None, result=None):
+        """Return previous comments by current bot and matching criteria."""
+        # Case-insensitive for backwards compatibility.
+        bot = self.__class__.__name__.lower()
```

Comment:
ic. I'll merge but I think there should be a follow up commit to have something like self.botname, in case some bot (or instance of one) needs to override this for whatever reason.